autogen/agentchat/groupchat.py

```diff
@@ -1264,11 +1264,10 @@ def resume(
             if not message_speaker_agent and message["name"] == self.name:
                 message_speaker_agent = self
 
-            # Add previous messages to each agent (except their own messages and the last message, as we'll kick off the conversation with it)
+            # Add previous messages to each agent (except the last message, as we'll kick off the conversation with it)
             if i != len(messages) - 1:
                 for agent in self._groupchat.agents:
-                    if agent.name != message["name"]:
-                        self.send(message, self._groupchat.agent_by_name(agent.name), request_reply=False, silent=True)
+                    self.send(message, self._groupchat.agent_by_name(agent.name), request_reply=False, silent=True)
```

Comment:
Sending messages to every agent might causes issues when message type is not accepted by all agents

For example, when agent A support tool call while agent B doesn't support tool call, sending A's tool call message to B will cause an exception when sending LLM request.

Reply:
> Perhaps we should have a separate discussion around it and think about possible solutions. 

That makes complete sense to me. It sounds like a significant (and potentially breaking) change by adding mechanisms in llmconfig to handle message compatibility and support multiple LLMs.

From what I know, some teams are currently refactoring the code, improving its quality, and making it production-ready. Their efforts should also provide more confidence when developing new features like this and addressing bugs. Maybe we can revisit the solution once their updates are complete?

Reply:
Yes, that sounds like a plan. As you are more aware of those things I'll leave it with you to bring up.

On a side note, my thoughts are that if the agents can contain the messages with tools, the client classes can be the points at which those messages are adjusted to accommodate whether the LLM supports it (e.g. the agents and messages don't have to change, only the client classes).

I'm thinking we can still proceed with this PR?

Reply:
Yeah sure, approved